### PR TITLE
feat: add wasi bindings crate

### PR DESCRIPTION
**Pull request by eduardomourar**
While we don't have a registry for the WIT definitions, it is nice to be able to reference the base world for preview2 through a crate.

### CODE REVIEW

File: bindings/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+mod bindings {
+    wit_bindgen::generate!({
+        path: "../wit",
+        world: "reactor",
+        std_feature,
+        // The generated definition of command will pull in std, so we are defining it
+        // manually below instead
+        skip: ["command", "get-preopens", "get-environment"],
```
Comment (pchickey) [inline bindings/src/lib.rs:8]: I don't think we want to skip these bindings - thats a special case in the adapter

File: bindings/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    });
+}
+
+pub use bindings::*;
```
Comment (pchickey) [inline bindings/src/lib.rs:12]: lets just run the macro at the top level rather than in a mod and pub use it all

File: bindings/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "bindings"
```
Comment (pchickey) [inline bindings/Cargo.toml:2]: morally this is just the `wasi` crate. could we just name it that, with `publish=false`? And I think in that case we want to version it `0.12.0+preview2`?

**Review by pchickey — APPROVED** [review]
Agree this is something we want! Excuse all the quibbles.

File: bindings/Cargo.toml (hunk 3 above)
Comment (eduardomourar) [reply bindings/Cargo.toml]: i thought so. i pushed all the changes based on your suggestion

**Review by pchickey — APPROVED** [review]

**Pull request merged**